### ISSUE TITLE
Recursive load and dependency error on develop branch.

### ISSUE THREAD

**Issue by Vvkmnn** [hlissner/doom-emacs#586]
Hey again,

### Observed behavior

So, I just pulled in 76f09b077bafce364d7e17e7e09bbc8c55badff5, and I'm seeing the following issues after a fresh `make install`:

![](https://i.imgur.com/RgBKP4y.png)

Doesn't seem like an issue, but the error literally prevents from opening files via `SPC .` failing every time on the `company-box` dependency. Not sure why. 

Ran `make doctor` and got this (pasted below): 

![](https://i.imgur.com/so0naKm.png)

### Expected behavior

Might be because I'm on High Sierra 10.13.5 Beta (17F70a), but I tried with the `emacs-plus` and `emacs-mac` variants, and saw no difference. For posterity, settled on the [emacs-mac](https://bitbucket.org/mituharu/emacs-mac/src/master/) cask in the end; looks nicer on an macOS. 
### Steps to reproduce

Sldo Ran a `brew cask install emacs-mac-spacemacs-icon` then `brew install emacs --with-gnutls --with-cocoa` after to get `make install` running. If not (or just the cask) would complain that `advice-add` was a void function:

```
# without `brew install emacs --with-gnutls --with-cocoa` 
# with `brew cask install emacs-mac`
Symbol's function definition is void: advice-add
make: *** [install] Error 255
```
### Hypothesis
Digging into the issues, it seems to have a lot to do with Emacs 26 supporting these childframe features, while Emacs 25 can't. Seems to be a similar case in https://github.com/hlissner/doom-emacs/issues/566, https://github.com/hlissner/doom-emacs/issues/470, https://github.com/hlissner/doom-emacs/issues/509. 

I tried to follow the instructions in https://github.com/hlissner/doom-emacs/issues/71#issuecomment-302750785, but sudo won't allow me to delete `usr/bin/emacs` with an `Operation not permitted`, which is odd in and of itself. xD

### System information

<details>
<summary>Click to expand</summary>

```
- OS: darwin (x86_64-apple-darwin13.4.0)
- Emacs: 26.1 (Apr 15, 2018)
- Doom: 2.0.9 (develop https://github.com/hlissner/doom-emacs/commit/76f09b077bafce364d7e17e7e09bbc8c55badff5)
- Graphic display: t (daemon: nil)
- System features: NOTIFY ACL LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS
- Details:
  ```elisp
  uname -a:  Darwin vBook 17.6.0 Darwin Kernel Version 17.6.0: Tue May  8 15:33:09 PDT 2018; root:xnu-4570.61.1~2/RELEASE_X86_64 x86_64
  modules:   (:feature eval (evil +everywhere) file-templates (lookup +devdocs +docsets) services snippets spellcheck (syntax-checker +childframe) version-control workspaces :completion (ivy +childframe) :ui doom doom-dashboard doom-modeline doom-quit evil-goggles hl-todo nav-flash neotree vi-tilde-fringe window-select :tools editorconfig ein electric-indent eshell gist imenu macos make magit password-store pdf rgb rotate-text term tmux upload :lang cc crystal clojure csharp data elixir elm emacs-lisp ess go (haskell +intero) hy (java +meghanada) javascript julia latex ledger lua markdown nim nix ocaml (org +attach +babel +capture +export +present +publish) perl php plantuml purescript python rest ruby rust scala sh swift web :app :config)
  packages:  (ace-link ace-window adaptive-wrap alchemist all-the-icons anaconda-mode android-mode anzu async auctex auto-compile auto-yasnippet avy (centered-window :recipe (centered-window :fetcher github :repo anler/centered-window-mode)) cider circe circe-notifications clj-refactor cmake-mode coffee-mode command-log-mode company company-anaconda company-auctex company-box company-dict company-erlang (company-glsl :recipe (company-glsl :fetcher github :repo Kaali/company-glsl)) company-go company-inf-ruby company-irony company-irony-c-headers company-lua company-php company-racer company-restclient company-shell company-sourcekit company-statistics company-web counsel counsel-css counsel-dash counsel-projectile crystal-mode csharp-mode cuda-mode demangle-mode (devdocs-lookup :recipe (devdocs-lookup :fetcher github :repo skeeto/devdocs-lookup)) dired-k disaster dockerfile-mode doom-themes dumb-jump editorconfig ein elfeed elfeed-org elixir-mode elm-mode (emacs-snippets :recipe (emacs-snippets :fetcher github :repo hlissner/emacs-snippets :files (*))) emmet-mode ensime erlang eslintd-fix ess ess-R-data-view ess-smart-equals evil evil-anzu evil-args evil-collection evil-commentary evil-easymotion evil-embrace evil-escape evil-exchange evil-goggles evil-indent-plus evil-ledger evil-magit evil-matchit evil-mc evil-multiedit evil-numbers evil-org evil-snipe evil-surround evil-textobj-anyblock evil-vimish-fold evil-visualstar (exec-path-from-shell :ignore nil) expand-region flycheck flycheck-cask flycheck-crystal flycheck-elm flycheck-irony flycheck-ledger flycheck-nim flycheck-perl6 flycheck-plantuml flycheck-popup-tip flycheck-posframe flycheck-rebar3 flycheck-rust flyspell flyspell-correct flyspell-correct-ivy gist git-gutter-fringe git-link git-timemachine gitconfig-mode gitignore-mode glsl-mode go-eldoc go-guru go-mode gorepl-mode graphql-mode groovy-mode haml-mode haskell-mode haxor-mode helpful hide-mode-line highlight-indentation highlight-numbers highlight-quoted hindent hl-todo htmlize hy-mode hydra imenu-anywhere imenu-list inf-crystal inf-ruby intero irony irony-eldoc ivy ivy-bibtex ivy-erlang-complete ivy-hydra ivy-posframe ivy-rich ivy-rtags ivy-xref js2-mode js2-refactor json-mode julia-mode kurecolor langtool ledger-mode less-css-mode lua-mode macrostep magit magithub makefile-executor markdown-mode markdown-toc meghanada merlin mips-mode mixed-pitch modern-cpp-font-lock moonscript mu4e-maildirs-extension multi-term nasm-mode nav-flash neotree nim-mode nix-mode nodejs-repl nose ob-crystal ob-go ob-mongo ob-nim ob-rust ob-sql-mode ob-translate omnisharp opencl-mode (org-bullets :recipe (org-bullets :fetcher github :repo Kaligule/org-bullets)) org-download org-plus-contrib org-tree-slide (osx-clipboard :ignore nil) overseer ox-pandoc ox-reveal pass password-store pcre2el pdf-tools perl6-mode persistent-soft persp-mode php-boris (php-extras :recipe (php-extras :fetcher github :repo arnested/php-extras)) php-mode php-refactor-mode phpunit pip-requirements plantuml-mode prodigy projectile psc-ide pug-mode purescript-mode quelpa quickrun racer rainbow-delimiters rainbow-mode rake realgud restart-emacs restclient rjsx-mode (rotate-text :recipe (rotate-text :fetcher github :repo debug-ito/rotate-text.el)) rspec-mode rtags ruby-refactor rust-mode sass-mode sbt-mode scala-mode shader-mode shrink-path skewer-mode slim-mode slime smartparens smex solaire-mode ssh-deploy stylus-mode swift-mode swiper synosaurus tabbar tide toc-org toml-mode tuareg twittering-mode typescript-mode undo-tree unicode-fonts use-package vi-tilde-fringe vimrc-mode visual-fill-column web-beautify web-mode wgrep which-key wordnut xref-js2 yaml-mode yard-mode yasnippet)
  elc dirs:  n/a
  exec-path: (/usr/local/bin/ /usr/bin/ /bin/ /usr/sbin/ /sbin/ /opt/X11/bin/ /Applications/Emacs.app/Contents/MacOS/libexec/)
```
</details>

Thanks!

**Comment by Vvkmnn** [comment]
Discovered temporary success by disabling any 'childframe' reference in `init.el`. Turning them off allows me to open and edit files again. 

```e-lisp

(doom! :feature
       debugger          ; FIXME stepping through code, to help you add bugs
       eval              ; run code, run (also, repls)
       (evil +everywhere); come to the dark side, we have cookies
       file-templates    ; auto-snippets for empty files
       (lookup           ; helps you navigate your code and documentation
        +devdocs         ; ...on devdocs.io online
        +docsets)        ; ...or in Dash docsets locally
       services          ; TODO managing external services & code builders
       snippets          ; my elves. They type so I don't have to
       spellcheck        ; tasing you for misspelling mispelling
       syntax-checker   ; tasing you for every semicolon you forget
        ; +childframe)     ; use childframes for error popups (Emacs 26+ only)
       version-control   ; remember, remember that commit in November
       workspaces        ; tab emulation, persistence & separate workspaces

       :completion
       (company          ; the ultimate code completion backend
        +auto)            ; as-you-type code completion
        ; +childframe)     ; a nicer company UI (Emacs 26+ only)
      ;helm              ; the *other* search engine for love and life
      ;ido               ; the other *other* search engine...
       ivy              ; a search engine for love and life
        ; +childframe)     ; uses childframes for popups (Emacs 26+ only)

       :ui
       (popup            ; tame sudden yet inevitable temporary windows
        +all             ; catch all popups that start with an asterix
        +defaults)       ; default popup rules
       doom              ; what makes DOOM look the way it does
       doom-dashboard    ; a nifty splash screen for Emacs
       doom-modeline     ; a snazzy Atom-inspired mode-line
       doom-quit         ; DOOM quit-message prompts when you quit Emacs
       evil-goggles      ; display visual hints when editing in evil
       hl-todo           ; highlight TODO/FIXME/NOTE tags
       nav-flash         ; blink the current line after jumping
       neotree           ; a project drawer, like NERDTree for vim
      ;tabbar            ; FIXME an (incomplete) tab bar for Emacs
      ;unicode           ; extended unicode support for various languages
       vi-tilde-fringe   ; fringe tildes to mark beyond EOB
       window-select     ; visually switch windows

       :tools
       dired             ; making dired pretty [functional]
       editorconfig      ; let someone else argue about tabs vs spaces
       ein               ; tame Jupyter notebooks with emacs
       electric-indent   ; smarter, keyword-based electric-indent
       eshell            ; a consistent, cross-platform shell (WIP)
       gist              ; interacting with github gists
       imenu             ; an imenu sidebar and searchable code index
       macos             ; MacOS-specific commands
       make              ; run make tasks from Emacs
       magit             ;
       password-store    ; password manager for nerds
       pdf               ; pdf enhancements
       rgb               ; creating color strings
       rotate-text       ; cycle region at point between text candidates
       term              ; terminals in Emacs
       tmux              ; an API for interacting with tmux
       upload            ; map local to remote projects via ssh/ftp

       :lang
       assembly          ; assembly for fun or debugging
       cc                ; C/C++/Obj-C madness
       crystal           ; ruby at the speed of c
       clojure           ; java with a lisp
       csharp            ; unity, .NET, and mono shenanigans
       data              ; config/data formats
      ;erlang            ; an elegant language for a more civilized age
       elixir            ; erlang done right
       elm               ; care for a cup of TEA?
       emacs-lisp        ; drown in parentheses
       ess               ; emacs speaks statistics
       go                ; the hipster dialect
       (haskell +intero) ; a language that's lazier than I am
       hy                ; readability of scheme w/ speed of python
       (java +meghanada) ; the poster child for carpal tunnel syndrome
       javascript        ; all(hope(abandon(ye(who(enter(here))))))
       julia             ; a better, faster MATLAB
       latex             ; writing papers in Emacs has never been so fun
       ledger            ; an accounting system in Emacs
       lua               ; one-based indices? one-based indices
       markdown          ; writing docs for people to ignore
       nim               ; python + lisp at the speed of c
       nix               ; I hereby declare "nix geht mehr!"
       ocaml             ; an objective camel
       (org              ; organize your plain life in plain text
        +attach          ; custom attachment system
        +babel           ; running code in org
        +capture         ; org-capture in and outside of Emacs
        +export          ; Exporting org to whatever you want
        +present         ; Emacs for presentations
        +publish)        ; Emacs+Org as a static site generator
       perl              ; write code no one else can comprehend
       php               ; perl's insecure younger brother
       plantuml          ; diagrams for confusing people more
       purescript        ; javascript, but functional
       python            ; beautiful is better than ugly
       rest              ; Emacs as a REST client
       ruby              ; 1.step do {|i| p "Ruby is #{i.even? ? 'love' : 'life'}"}
       rust              ; Fe2O3.unwrap().unwrap().unwrap().unwrap()
       scala             ; java, but good
       sh                ; she sells (ba|z)sh shells on the C xor
       swift             ; who asked for emoji variables?
       typescript        ; javascript, but better
       web               ; the tubes

       ;; Applications are complex and opinionated modules that transform Emacs
       ;; toward a specific purpose. They may have additional dependencies and
       ;; should be loaded late.
       :app
      ;(email +gmail)    ; emacs as an email client
      ;irc               ; how neckbeards socialize
      ;(rss +org)        ; emacs as an RSS reader
      ;twitter           ; twitter client https://twitter.com/vnought
      (write            ; emacs as a word processor (latex + org + markdown)
       +wordnut         ; wordnet (wn) search
       +langtool)       ; a proofreader (grammar/style check) for Emacs

       :collab
      ;impatient-mode    ; show off code over HTTP

       :config
       ;; The default module set reasonable defaults for Emacs. It also provides
       ;; a Spacemacs-inspired keybinding scheme, a custom yasnippet library,
       ;; and additional ex commands for evil-mode. Use it as a reference for
       ;; your own modules.
       (default +bindings +snippets +evil-commands))
```

Make doctor is still being silly though:
```
Doom Doctor
Emacs v25.3.1
Doom v2.0.9 (b4a7e534)
shell: /bin/zsh
Compiled with:
  JPEG NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS
uname -a:
  Darwin vBook 17.6.0 Darwin Kernel Version 17.6.0: Tue May 8 15:33:09 PDT 2018;
  root:xnu-4570.61.1~2/RELEASE_X86_64 x86_64
----
test-emacs
test-private-config
test-windows
test-fonts
test-gnutls
test-tls
Validated https://elpa.gnu.org
Validated https://melpa.org
Successfully rejected https://self-signed.badssl.com
Successfully rejected https://wrong.host.badssl.com/
test-tar
----
Doom loaded 336 packages across 0 modules in 3.681s
Recursive load: "/Users/v/.emacs.d/core/autoload/buffers.el", "/Users/v/.emacs.d/core/autoload/buffers.el", "/Users/v/.emacs.d/core/autoload/buffers.el", "/Users/v/.emacs.d/core/autoload/buffers.el", "/Users/v/.emacs.d/core/autoload/buffers.el", "/Users/v/.emacs.d/bin/doom-doctor"
make: *** [doctor] Error 255
```

**Comment by hlissner** [comment]
A great many changes have been made in the past week to the `develop` branch, and I am preparing for the 2.1 release. Could you update and let me know if this problem still exists?

I have yet to get my hands on a High Sierra VM to test Doom in, and am struggling to do so (I am currently traveling, as well), so I'd appreciate any help I can get.

**Comment by Vvkmnn** [comment]
I'm sure it's just a quirk of macOS High Sierra. I've been flying [Spacemacs](http://spacemacs.org/) for a few days now and it does the job, but happy / eager to switch back to Doom after I ship this project out. 

I'll re-run the config from the above example with the latest commit from `develop` later this week, and post my results here. 👍

**Comment by Vvkmnn** [comment]
Just pulled a77a967299fb7fac4cff240aa9d63c5d799b89bd off of develop onto macOS High Sierra 10.13.5 Beta (17F70a) , and ran `make quickstart` (after deleting `.local/` & `.cache/`). 

This created a `~/.doom.d/init.el` that I modified to: 
```e-lsip
;;; init.el -*- lexical-binding: t; -*-
;; Copy me to ~/.doom.d/init.el or ~/.config/doom/init.el, then edit me!

(doom! :feature
       debugger          ; stepping through code, to help you add bugs
       eval              ; run code, run (also, repls)
       (evil +everywhere); come to the dark side, we have cookies
       file-templates    ; auto-snippets for empty files
       (lookup           ; helps you navigate your code and documentation
        +devdocs         ; ...on devdocs.io online
        +docsets)        ; ...or in Dash docsets locally
       snippets          ; my elves. They type so I don't have to
       spellcheck        ; tasing you for misspelling mispelling
       (syntax-checker   ; tasing you for every semicolon you forget
        +childframe)     ; use childframes for error popups (Emacs 26+ only)
       version-control   ; remember, remember that commit in November
       workspaces        ; tab emulation, persistence & separate workspaces

       :completion
       (company          ; the ultimate code completion backend
        +auto)           ; as-you-type code completion
      ;(helm             ; the *other* search engine for love and life
      ; +fuzzy)          ; enable fuzzy search backend for helm
      ;ido               ; the other *other* search engine...
       (ivy              ; a search engine for love and life
        +fuzzy)          ; enable fuzzy search backend for ivy

       :ui
       doom              ; what makes DOOM look the way it does
       doom-dashboard    ; a nifty splash screen for Emacs
       doom-modeline     ; a snazzy Atom-inspired mode-line
       doom-quit         ; DOOM quit-message prompts when you quit Emacs
       evil-goggles      ; display visual hints when editing in evil
      ;fci               ; a `fill-column' indicator
       hl-todo           ; highlight TODO/FIXME/NOTE tags
       nav-flash         ; blink the current line after jumping
      ;neotree           ; a project drawer, like NERDTree for vim
       treemacs          ; a project drawer, like neotree but cooler
       (popup            ; tame sudden yet inevitable temporary windows
        +all             ; catch all popups that start with an asterix
        +defaults)       ; default popup rules
       tabbar            ; FIXME an (incomplete) tab bar for Emacs
       unicode           ; extended unicode support for various languages
       vi-tilde-fringe   ; fringe tildes to mark beyond EOB
       window-select     ; visually switch windows

       :emacs
       dired             ; making dired pretty [functional]
       ediff             ; comparing files in Emacs
       electric-indent   ; smarter, keyword-based electric-indent
      ;eshell            ; a consistent, cross-platform shell (WIP)
       imenu             ; an imenu sidebar and searchable code index
       term              ; terminals in Emacs

       :tools
       editorconfig      ; let someone else argue about tabs vs spaces
       ein               ; tame Jupyter notebooks with emacs
       gist              ; interacting with github gists
       macos             ; MacOS-specific commands
       make              ; run make tasks from Emacs
       magit             ;
      ;password-store    ; password manager for nerds
       pdf               ; pdf enhancements
       prodigy           ; managing external services & code builders
       rgb               ; creating color strings
       rotate-text       ; cycle region at point between text candidates
      ;tmux              ; an API for interacting with tmux
      ;upload            ; map local to remote projects via ssh/ftp
      ;wakatime

       :lang
      ;assembly          ; assembly for fun or debugging
      ;cc                ; C/C++/Obj-C madness
      ;crystal           ; ruby at the speed of c
      ;clojure           ; java with a lisp
      ;csharp            ; unity, .NET, and mono shenanigans
       data              ; config/data formats
      ;erlang            ; an elegant language for a more civilized age
      ;elixir            ; erlang done right
      ;elm               ; care for a cup of TEA?
       emacs-lisp        ; drown in parentheses
      ;ess               ; emacs speaks statistics
      ;go                ; the hipster dialect
      ;(haskell +intero) ; a language that's lazier than I am
      ;hy                ; readability of scheme w/ speed of python
      ;(java +meghanada) ; the poster child for carpal tunnel syndrome
       javascript        ; all(hope(abandon(ye(who(enter(here))))))
      ;julia             ; a better, faster MATLAB
      ;latex             ; writing papers in Emacs has never been so fun
      ;ledger            ; an accounting system in Emacs
      ;lua               ; one-based indices? one-based indices
       markdown          ; writing docs for people to ignore
      ;nim               ; python + lisp at the speed of c
      ;nix               ; I hereby declare "nix geht mehr!"
      ;ocaml             ; an objective camel
       (org              ; organize your plain life in plain text
        +attach          ; custom attachment system
        +babel           ; running code in org
        +capture         ; org-capture in and outside of Emacs
        +export          ; Exporting org to whatever you want
        +present)        ; Emacs for presentations
      ;perl              ; write code no one else can comprehend
      ;php               ; perl's insecure younger brother
      ;plantuml          ; diagrams for confusing people more
      ;purescript        ; javascript, but functional
      ;python            ; beautiful is better than ugly
      ;qt                ; the 'cutest' gui framework ever
      ;rest              ; Emacs as a REST client
      ;ruby              ; 1.step do {|i| p "Ruby is #{i.even? ? 'love' : 'life'}"}
      ;rust              ; Fe2O3.unwrap().unwrap().unwrap().unwrap()
      ;scala             ; java, but good
       sh                ; she sells (ba|z)sh shells on the C xor
      ;solidity          ; do you need a blockchain? No.
      ;swift             ; who asked for emoji variables?
       web               ; the tubes

       ;; Applications are complex and opinionated modules that transform Emacs
       ;; toward a specific purpose. They may have additional dependencies and
       ;; should be loaded late.
       :app
      ;(email +gmail)    ; emacs as an email client
      ;irc               ; how neckbeards socialize
      ;(rss +org)        ; emacs as an RSS reader
      ;twitter           ; twitter client https://twitter.com/vnought
       (write            ; emacs as a word processor (latex + org + markdown)
        +wordnut         ; wordnet (wn) search
        +langtool)       ; a proofreader (grammar/style check) for Emacs

       :collab
      ;floobits          ; peer programming for a price
      ;impatient-mode    ; show off code over HTTP

       :config
       ;; For literate config users. This will tangle+compile a config.org
       ;; literate config in your `doom-private-dir' whenever it changes.
      ;literate

       ;; The default module set reasonable defaults for Emacs. It also provides
       ;; a Spacemacs-inspired keybinding scheme, a custom yasnippet library,
       ;; and additional ex commands for evil-mode. Use it as a reference for
       ;; your own modules.
       (default +bindings +snippets +evil-commands))

```

Running `make install` ran with no errors, but hangs for a while. Little whining about some forms and `evil-magit`, but I had to force quit on clearing recentf. 
```zsh
...
Installing coffee-mode
✓ DONE
Finished!
✓ Cached package state
✓ Package autoloads included
✓ Removed load-path/auto-mode-alist entries

In toplevel form:
../../../autoloads.pkg.el:354:184:Warning: ‘org-add-link-type’ is an obsolete
    function (as of Org 9.0); use ‘org-link-set-parameters’ instead.
evil-magit initialized

Cleaning up the recentf list...Cleaning up the recentf list...done (0 removed)
```

Ran again with a `make` with no errors. Everything was going great, until I hit `SPC SPC`, and the entire thing started to hang:

![image](https://user-images.githubusercontent.com/1954615/41311474-5f35d27a-6e52-11e8-86e6-346c3f15ca6e.png)

Basically had to force restart after this, but other key bindings seem to work so far! Nice job bud.

**Comment by Vvkmnn** [comment]
This is a seperate issue, so I'm gonna close this and try to write it up as an another issue / find it in someone elses.

Thanks!
